sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -822,7 +823,9 @@ class Analyzer(
           .getOrElse(alter)
 
       case u: UnresolvedV2Relation =>
-        CatalogV2Util.loadRelation(u.catalog, u.tableName).getOrElse(u)
+        CatalogV2Util.loadRelation(u.catalog, u.tableName)
+          .map(SubqueryAlias(u.originalNameParts, _))
```

Comment:
do we need to add `SubqueryAlias` for UnresolvedV2Relation? AFAIK it's only for ALTER TABLE, which does not need to resolve qualified column names.

Reply:
reverted this change.